subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppApplicationPublishingIntegrationTest.groovy

```diff
@@ -164,6 +169,15 @@ class CppApplicationPublishingIntegrationTest extends AbstractCppPublishingInteg
                 dependencies {
                     implementation project(':greeter')
                 }
+                application.binaries.configureEach {
+                    // TODO: See https://github.com/gradle/gradle-native/issues/142
+                    linkTask.get().linkerArgs.addAll(providers.provider {
+                        if (targetPlatform.operatingSystemFamily.linux) {
+                            return ["-Wl,-rpath,'\\\$ORIGIN'"]
```

Comment:
We should change this test, as it's not really verifying a supported behaviour. We only really support using an install image rather than using the executable (we might add support later though). So, we should change the publishing for an application to publish the install image instead of the executable and change this test to resolve and unzip the install image and try to use it after running clean. The result should be that the app should work without any special compiler flags required in the DSL.

Reply:
Ah, that's what I was missing.  I thought we were running from the install still.

I think publishing an install image is a good idea, but we shouldn't let this blow out of scope. Changing the application plugin to publish a distribution/installation requires a little more planning (new component? new tasks on CppExecutable? etc).  I opened a new issue for it.

@lacasseio can we just change the `install` task in the consumer build to be a `InstallExecutable` configured properly to build a new installation based on the published executable (+dependencies)?